war/src/test/java/io/lumeer/engine/rest/UserGroupServiceIntegrationTest.java

```diff
@@ -0,0 +1,251 @@
+      final Client client = ClientBuilder.newBuilder().build();
+
+      String group1 = "AddGroup1";
+      String group2 = "AddGroup2";
+
+      client.target(TARGET_URI).path(PATH_PREFIX)
+            .request(MediaType.APPLICATION_JSON)
+            .buildPost(Entity.entity(Arrays.asList(group1, group2), MediaType.APPLICATION_JSON)).invoke();
+
+      List<String> groupsFromFacade = userGroupFacade.getGroups(organization);
+
+      assertThat(groupsFromFacade).containsOnly(group1, group2);
+   }
+
+   @Test
+   public void testAddUserToGroups() throws Exception {
+      final Client client = ClientBuilder.newBuilder().build();
+
+      String user = "user";
+      userGroupFacade.addUser(organization, user);
+      String group1 = "AddUserToGroup1";
+      String group2 = "AddUserToGroup2";
+      userGroupFacade.addGroups(organization, group1, group2);
+
+      client.target(TARGET_URI).path(PATH_PREFIX + "users/" + user)
+            .request(MediaType.APPLICATION_JSON)
+            .buildPut(Entity.entity(Arrays.asList(group1, group2), MediaType.APPLICATION_JSON)).invoke();
+
+      List<String> groupsFromFacade = userGroupFacade.getGroupsOfUser(organization, user);
+
+      assertThat(groupsFromFacade).containsOnly(group1, group2);
+   }
+
+   @Test
+   public void testRemoveUser() throws Exception {
+      final Client client = ClientBuilder.newBuilder().build();
+
+      String user = "user";
+      userGroupFacade.addUser(organization, user);
+
+      client.target(TARGET_URI).path(PATH_PREFIX + "users/" + user)
+            .request(MediaType.APPLICATION_JSON)
+            .buildDelete().invoke();
+
+      assertThat(userGroupFacade.getUsers(organization)).isEmpty();
+   }
+
+   @Test
+   public void testRemoveUserFromGroups() throws Exception {
+      final Client client = ClientBuilder.newBuilder().build();
+
+      String user = "user";
+      String group1 = "RemoveUserFromGroups1";
+      String group2 = "RemoveUserFromGroups2";
+      userGroupFacade.addGroups(organization, group1, group2);
+      userGroupFacade.addUser(organization, user, group1, group2);
+
+      client.target(TARGET_URI).path(PATH_PREFIX + "users/" + user + "/groups")
+            .request(MediaType.APPLICATION_JSON)
+            .buildPut(Entity.entity(Arrays.asList(group1), MediaType.APPLICATION_JSON)).invoke();
+
+      assertThat(userGroupFacade.getGroupsOfUser(organization, user)).containsOnly(group2);
+   }
+
+   // TODO: reading map from response does not work (JsonMappingException) and I don't know why"
+   @Test
+   public void testGetUsersAndGroups() throws Exception {
+      final Client client = ClientBuilder.newBuilder().build();
+
+      Response response = client.target(TARGET_URI).path(PATH_PREFIX + "users").request(MediaType.APPLICATION_JSON).buildGet().invoke();
+      Map<String, List<String>> usersAndGroups = response.readEntity(Map.class); // does not work
+      assertThat(usersAndGroups).isEmpty();
```

Comment:
I've seen it in some test but it didn't work... I suppose it's because there is a collection in collection?